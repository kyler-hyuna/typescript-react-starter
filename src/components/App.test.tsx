import React from 'react'
import { Provider } from 'react-redux'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import ConnectedApp, { App } from './App'

const store = configureMockStore()({ counter: 0 })

it('it should work', () => {
  const tree = create(
    <Provider store={store}>
      <ConnectedApp />
    </Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('should just render', () => {
  const wrapper = shallow(<App counter={0} increment={jest.fn()} />)
  expect(wrapper.find('h1').text()).toBe('0 new react projects')
})
