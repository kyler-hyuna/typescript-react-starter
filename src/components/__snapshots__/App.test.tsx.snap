// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should work 1`] = `
Array [
  <h1>
    0
     new react projects
  </h1>,
  <p>
    na
  </p>,
  <p>
    ma
  </p>,
  <img
    alt=""
    src="test-file-stub"
  />,
  <button
    onClick={[Function]}
  >
    Hey
  </button>,
]
`;
